### PR TITLE
#6591: Add optional output tensor for moreh sum forward and backward

### PR DESCRIPTION
**Pull request by bharane-ab**

### CODE REVIEW

**Review by eyonland — CHANGES_REQUESTED** [review]
Please add a test showing that the optional output tensor does not even need to be provided.  And one that shows that the optional output tensor was provided and returned back by the op.

File: tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_op.cpp [hunk 1]
```diff
@@ -108,30 +108,44 @@ Tensor moreh_sum_(const Tensor& input, const int64_t& dim, const MemoryConfig& m
     return output;
 }
 
-Tensor moreh_sum(
+std::vector<Tensor> moreh_sum(
     const Tensor& input,
```
Comment (eyonland) [inline tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_op.cpp:112]: I don't follow why this is being change to a vector?

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_sum.py [hunk 2]
```diff
@@ -79,29 +86,52 @@ def get_backward_tensors(output_grad_shape, input_grad_shape, device):
...
 
-    (tt_input, tt_output, torch_input) = get_tensors(input_shape, output_shape, device)
+    cpu_layout = ttl.tensor.Layout.ROW_MAJOR
+
+    if optional_tensor:
+        (tt_input, tt_output, torch_input, optional_output, optional_torch) = get_tensors(
+            input_shape, output_shape, optional_tensor, device
+        )
+
+        tt_output_cpu = ttl.operations.primary.moreh_sum(tt_input, tt_output, dims=dims, output_tensor=optional_output)
```
Comment (eyonland) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_sum.py:110]: Is this really a tensor on cpu?

**Review by eyonland — CHANGES_REQUESTED** [review]

**Comment by bharane-ab** [comment]
Moreh team is addressing the issue

**Pull request closed without merging**